spec/unread/readable_spec.rb

```diff
@@ -266,6 +266,17 @@
       expect(@email2.unread?(@reader)).to be_falsey
     end
 
+    it "should mark the rest as read when the first record is not unique" do
+      Email.mark_as_read! [ @email1 ], for: @reader
+
+      allow(@email1).to receive_message_chain("read_marks.build").and_return(@email1.read_marks.build)
+      allow(@email1).to receive_message_chain("read_marks.where").and_return([])
```

Comment:
Instead of mocking the implementation: What do you think of using two threads to demonstrate that it's really thread-safe now?